CSLabs.Api/Email/SESSender.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Amazon;
+using System.Linq;
+using Amazon.SimpleEmailV2;
+using Amazon.SimpleEmailV2.Model;
+using System.Threading;
+using FluentEmail.Core;
+using FluentEmail.Core.Interfaces;
+using FluentEmail.Core.Models;
+
+namespace CSLabs.Api.Email
+{
+    public class SESSender : ISender
+    {
+        private readonly string _accessKey;
+        private readonly string _secretKey;
+        private readonly RegionEndpoint _endpoint;
+
+        public SESSender(string accessKey, string secretKey, RegionEndpoint endpoint)
+        {
+            _accessKey = accessKey;
+            _secretKey = secretKey;
+            _endpoint = endpoint;
+        }
+        
+        public SendResponse Send(IFluentEmail email, CancellationToken? token = null)
+        {
+            return SendAsync(email, token).GetAwaiter().GetResult();
+        }
+
+        public async Task<SendResponse> SendAsync(IFluentEmail email, CancellationToken? token = null)
+        {
+            using var client = new AmazonSimpleEmailServiceV2Client(_accessKey, _secretKey, _endpoint);
+            var sendRequest = new SendEmailRequest
+            {
+                FromEmailAddress = email.Data.FromAddress.EmailAddress,
+                Destination = new Destination
+                {
+                    ToAddresses = email.Data.ToAddresses.Select(e => e.EmailAddress).ToList()
+                },
+                Content = new EmailContent
+                {
+                    Simple = new Message
+                    {
+                        Subject = new Content
+                        {
+                            Charset = "UTF-8",
+                            Data = email.Data.Subject
+                        },
+                        Body = new Body
+                        {
+                            Html = new Content
+                            {
+                                Charset = "UTF-8",
+                                Data = email.Data.Body
+                            }
+                        }
+                    }
+                },
+                ReplyToAddresses = new List<string> {"iuscompsec@gmail.com"}
```

Comment:
Can we have this replyToAddress configurable in the appsettings?

Reply:
I did that originally. I would have to pass the replyToAddress in the EmailExtensions.cs for all emails since I cannot access the appsettings directly here. That's what I should do, correct? Or, I can also pass it from the AddSESSender, but I thought that would add unrelated details for that.

Reply:
I see your point. The ReplyTo is a part of the IFluentEmail interface and thus should be passed in that way.

You can do this by overriding the IFluentEmailFactory with your own factory and binding it to dependency injection in the Startup.cs file.

Create `class AppFluentEmailFactory : IFluentEmailFactory`

```
services.AddTransient<IFluentEmailFactory, AppFluentEmailFactory >();
```